graphql/batch_executor.go

```diff
@@ -131,9 +131,19 @@ func executeBatchWorkUnit(unit *WorkUnit) []*WorkUnit {
 		}
 		return nil
 	}
-	unitChildren, err := resolveBatch(unit.ctx, results, unit.field.Type, unit.selection.SelectionSet, unit.destinations)
+	nonNilSrcs := make([]interface{}, 0, len(results))
```

Comment:
why doesn't this get handled in eg the executing of an object or value? i worry that we have multiple places handling nil, and so eg also multiple places handling non-nullable fields.

Reply:
I can make that change.